### PR TITLE
1195 enhanced coverage util also triggers doc query

### PR DESCRIPTION
**Pull request by leite08**
Ref: metriport/metriport-internal#1195

### Dependencies

none

### Description

- enhanced coverage util also triggers doc query
- adds script to run lambda from local/utils
- improve executeAsynchronously()
- add OS search time to prod's dash

### Release Plan

- nothing special, mostly utils

### CODE REVIEW

File: packages/core/src/util/concurrency.ts [hunk 1]
```diff
@@ -49,31 +49,29 @@ export type FunctionType<T> = (
...
- *    start at slight different times. Defaults to no jitter.
  */
 export async function executeAsynchronously<T>(
   collection: T[],
   fn: FunctionType<T>,
   {
     numberOfParallelExecutions = collection.length,
     maxJitterMillis = 0,
-    jitterPct,
+    minJitterMillis = 0,
```
Comment (leite08) [inline packages/core/src/util/concurrency.ts:59]: This makes this function more easy to use, the "pct" was confusing.

File: packages/infra/dashboards/cw-dashboard-prod.json
Comment (leite08) [inline packages/infra/dashboards/cw-dashboard-prod.json:1]: Added a new widget to prod's dash so we can see the amount of requests to OpenSearch

File: packages/lambdas/src/cw-link-patients.ts [hunk 2]
```diff
@@ -66,7 +66,7 @@ export const handler = Sentry.AWSLambda.wrapHandler(async (event: SQSEvent) => {
 
     console.log(`Done.`);
   }
-  console.log(`Total time: ${Date.now() - startedAt}ms`);
+  console.log(`Total time: ${Date.now() - startedAt} ms`);
```
Comment (leite08) [inline packages/lambdas/src/cw-link-patients.ts:69]: Updated EC's script so its easier to copy the duration w/ double click on terminal, replaced others as well

File: packages/core/src/util/concurrency.ts [hunk 3]
```diff
@@ -49,31 +49,29 @@ export type FunctionType<T> = (
  * @param collection array of elements to be processed
  * @param fn the function to be executed asynchronously
  * @param options additional settings
```
Comment (Goncharo) [inline packages/core/src/util/concurrency.ts:51]: nit: in my C++ days, we would put the "preconditions" in the function header, so the calling programmer understand how to use the function at a glance - without needing to read the code 

for example, in this case we fail on different conditions in the `minJitterMillis` checks, but don't mention it in the header

File: packages/utils/src/lambda/execute-lambda.ts
Comment (leite08) [inline packages/utils/src/lambda/execute-lambda.ts:1]: Had this around and Ramil's ticke to fix the 404 notification from lambdas reminded me to push this so others can use as well

File: packages/utils/src/commonwell/doc-query-shared.ts [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+    maxDocQueryAttemts,
+    minDocsToConsiderCompleted,
+  } = config;
+  let docQueryAttempts = 0;
+  let docsFound = 0;
+  let queryComplete = false;
+
+  const metriportAPI = new MetriportMedicalApi(apiKey, {
+    baseAddress: apiUrl,
+  });
```
Comment (Goncharo) [inline packages/utils/src/commonwell/doc-query-shared.ts:57]: would leave a TODO for @jonahkaye to address under his PR related to the WH override https://github.com/metriport/metriport/pull/1106

**Review by RamilGaripov — APPROVED** [review]

**Review by Goncharo — APPROVED** [review]
left a couple comments to consider, otherwise good to :shipit:

File: packages/core/src/util/concurrency.ts (hunk 3 above)
Comment (leite08) [reply packages/core/src/util/concurrency.ts:51]: I documented it on its own type so when you hover the mouse over it you see the docs for the property (only adding to the function it doesn't work). I think that's the right place to add it, so we also are not duplicating stuff - worse than no doc is stale/wrong doc.

Cool?

File: packages/utils/src/commonwell/doc-query-shared.ts (hunk 4 above)
Comment (leite08) [reply packages/utils/src/commonwell/doc-query-shared.ts:57]: But we want to trigger WH in this case, right?

Comment (leite08) [reply packages/utils/src/commonwell/doc-query-shared.ts]: Added a param

**Pull request merged**